### PR TITLE
Enhance speed benchmark

### LINKED ISSUE #4500: Enhance speed benchmark
### Motivation

Currently we don't have speed benchmarks for
- [x] `JournalStorage`
- [ ] multi-objective optimization algorithms (`NSGAIISampler` and multi-objective version of `TPESampler`).

### Suggestion

Add those speed benchmarks.

We need to add suitable benchmark script like https://github.com/optuna/optuna/blob/master/benchmarks/asv/optimize.py and possibly update https://github.com/optuna/optuna/blob/master/asv.conf.json

### Additional context (optional)

_No response_

### PR DESCRIPTION
**Pull request by g-tamaki**
Resolve #4500

## Motivation

Currently we don't have speed benchmarks for
multi-objective optimization algorithms (NSGAIISampler and multi-objective version of TPESampler).

## Description of the changes

1. Define multi_objective function
2. Added parameters for multi_objective benchmark

Benchmarks were taken here.
https://github.com/g-tamaki/optuna/actions/runs/6404021061/job/17383726285

### CODE REVIEW

**Comment by HideakiImamura** [comment]
@g-tamaki Could you fix the CI error? The type checker (mypy) fails. I think we should change the type annotation of the returned value of the `multi_objective` to `tuple[float, ...]`.

File: benchmarks/asv/optimize.py [hunk 1]
```diff
@@ -44,30 +47,49 @@ def objective(self, trial: optuna.Trial) -> float:
...
+        x = trial.suggest_float("x", -100, 100)
+        y = trial.suggest_int("y", -100, 100)
+        return (x**2 + y**2, (x - 2) ** 2 + (y - 2) ** 2)
+
+    def optimize(
+        self, storage_mode: str, sampler_mode: str, n_trials: int, objective_type: str
+    ) -> None:
         with StorageSupplier(storage_mode) as storage:
             sampler = create_sampler(sampler_mode)
             study = optuna.create_study(storage=storage, sampler=sampler)
```
Comment (not522) [inline benchmarks/asv/optimize.py:60]: Could you specify `directions` for multi-objective? Currently, all trials are failed.

**Review by not522 — COMMENTED** [review]
Thank you for your PR! Please check my comments.

File: benchmarks/asv/optimize.py (hunk 1 above)
Comment (HideakiImamura) [reply benchmarks/asv/optimize.py:60]: You can specify the `directions` as follows.
```suggestion
            if objective_type == "single":
                directions = ["minimize"]
            elif objective_type == "multi":
                directions = ["minimize", "minimize"]
            else:
                assert False
            study = optuna.create_study(storage=storage, sampler=sampler, directions=directions)
```

Comment (g-tamaki) [reply benchmarks/asv/optimize.py]: Thank you for your advice.

**Review by not522 — APPROVED** [review]
Thank you for the update! LGTM!

**Review by HideakiImamura — APPROVED** [review]
LGTM.

**Pull request merged**